### PR TITLE
Code quality

### PR DESCRIPTION
**Pull request by Taeir**
## Changes
- Removed IPositional
- Reordered methods and fields
- Changed many javadocs
- Renamed fields, methods and method parameters to better reflect what they do.
- Improved performance by not constantly constructing and deconstructing Rectangles.
- Split up some methods

Note: when a field is marked transient, it is not serialized.

<layout, naming, documentation, performance

### CODE REVIEW

File: src/test/java/com/github/fishio/power_ups/TestPuFreeze.java [hunk 1]
```diff
@@ -3,7 +3,9 @@
 import static org.junit.Assert.assertEquals;
 
 import java.util.ArrayList;
```
Comment (Diocruel) [inline src/test/java/com/github/fishio/power_ups/TestPuFreeze.java]: _Auto-generated comment_

```
Avoid unused imports such as 'java.util.ArrayList'
```

You missed 3 unused imports here. ;)

<layout

**Comment by Diocruel** [comment]
All in all this seems to be a very good pull request: It greatly improves the readability of the code, improves the multiplayer server-client communication and refactored some smells! I guess you already fixed the comment I gave earlier in your other pull request.

And I learned what transient means :). 

<organisation, performance, naming, documentation

**Comment by Taeir** [comment]
@Diocruel I removed the unused imports.

<review

**Comment by Diocruel** [comment]
Okay, I find this pr good to merge then. Needs one more comment though.
<layout

**Comment by LMiljak** [comment]
Looks good, I'm merging it :+1: 

<layout, naming, documentation, performance

**Pull request merged**